client/vehicle.lua

```diff
@@ -0,0 +1,245 @@
+            Wait(sleep)
+        end
+    end)
+
+    CreateThread(function()
+        local alert = 0
+        local sleep
+        local showingHud = true
+        while cache.vehicle do
+            local data
+            local engineIsRunning = getIsVehicleEngineRunning(cache.vehicle)
+            if showingHud and not engineIsRunning then
+                data = {
+                    {
+                        type = 'vehiclehud',
+                        show = false
+                    }
+                }
+                showingHud = false
+            end
+
+            sleep = 1000
+            if engineIsRunning and isMinimapRendering() then
+                showingHud = true
+                local _, highbeam, lowbeam = getVehicleLightsState(cache.vehicle)
+                data = {
+                    {
+                        type = 'vehiclehud',
+                        show = true
+                    },
+                    {
+                        type = 'speed',
+                        speed = getEntitySpeed(cache.vehicle) * config.speedMultiplier
+                    },
+                    {
+                        type = 'gauge',
+                        name = 'fuel',
+                        value = getVehicleFuelLevel(cache.vehicle) or 100
+                    },
+                    {
+                        type = 'dashboardlights',
+                        highbeam = highbeam,
+                        lowbeam = lowbeam,
+                    }
+                }
+
+
+                if config.lowFuelAlert and getVehicleFuelLevel(cache.vehicle) < config.lowFuelAlert then
+                    if alert > 0 then
+                        alert = alert - 1
+                    else
+                        alert = 1500
+                        qbx.playAudio({
+                            audioName = "CONFIRM_BEEP",
+                            audioRef = 'HUD_MINI_GAME_SOUNDSET',
+                            source = cache.vehicle
+                        })
+                        exports.qbx_core:Notify(locale("notify.low_fuel"), "error")
+                    end
+                end
+                sleep = 100
+            end
+
+            if data then
+                sendNUIMessage({
+                    update = true,
+                    data = data
+                })
+            end
+            Wait(sleep)
+        end
+    end)
+end
+
+local function initVehicleHud()
+    local data = {
+        {
+            type = 'vehiclehud',
+            show = false
+        }
+    }
+
+    if cache.seat == -1 then
+        local nitroLevel = Entity(cache.vehicle).state.nitro or 0
+        data = {
+            {
+                type = 'gauge',
+                name = 'nitro',
+                value = nitroLevel,
+                show = nitroLevel > 0
+            },
+            {
+                type = 'speedmax',
+                speed = GetVehicleEstimatedMaxSpeed(cache.vehicle) * 5.6 -- meh idk

```

Comment:
why the meh idk

Reply:
It's a kinda random value, just noticed i should check for the mph/kmh too, but yeah it's just trying to be a bit above the estimated max speed, which can and mostly isn't the actual max speed